### PR TITLE
Updated with a better error message for remote object references

### PR DESCRIPTION
**Pull request by basasairohan**
### Description

Currently, when referencing a remote object with a 4-part object name, the message 'servername' is not currently supported in Babelfish is raised which is not clear. This commit changes the message to 'Remote object reference with 4-part object name' is not currently supported by Babelfish.

### Testing

- Use case based: Added a new test `babel-2646`
- Boundary conditions: N/A
- Arbitrary inputs: N/A
- Negative test scenarios: N/A
- Minor version upgrade tests: N/A
- Major version upgrade tests: N/A
- Performance tests: Have not seen big impact on SLA.

 
### Issues Resolved

BABEL-2646

Authored-by: Sai Rohan Basa [bsrohan@amazon.com](mailto:bsrohan@amazon.com)
Signed-off-by: Sai Rohan Basa [bsrohan@amazon.com](mailto:bsrohan@amazon.com)

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp [hunk 1]
```diff
@@ -1360,7 +1360,7 @@ antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitFunc_proc_name_server_data
 antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitFull_object_name(TSqlParser::Full_object_nameContext *ctx)
 {
 	if (ctx->DOT().size() >= 3 && ctx->server) /* server.db.schema.funcname */
-		handle(INSTR_UNSUPPORTED_TSQL_SERVERNAME_IN_NAME, "servername", getLineAndPos(ctx));
+		handle(INSTR_UNSUPPORTED_TSQL_SERVERNAME_IN_NAME, "Remote object reference with 4-part object name", getLineAndPos(ctx));
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp:1363]: We should also throw this new error message if someone tries executing a procedure through four-part object names. Check line 1349.

**Review by thephantomthief — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp (hunk 1 above)
Comment (basasairohan) [reply contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp:1363]: okay

File: test/JDBC/expected/babel-2646.out [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- servername (not supported)
+insert into yourserver.master.dbo.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: 'Remote object reference with 4-part object name' is not currently supported in Babelfish)~~
+
```
Comment (thephantomthief) [inline test/JDBC/expected/babel-2646.out]: Please add test case for four part procedure name as well

**Review by thephantomthief — DISMISSED** [review]

**Review by kuntalghosh — COMMENTED** [review]
Let's not introduce new test files with JIRA id. Try to find a relevant a file and the same test.

**Review by thephantomthief — APPROVED** [review]

**Review by thephantomthief — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**